### DISCUSSION TITLE
Plans to move code base to TypeScript?

### DISCUSSION THREAD

**Discussion by SeanPedersen** [lbragile/TabMerger#16] [General]
What do you think about TypeScript? IMO it could speed up future refactoring and help new devs make changes to the code.

**Comment by lbragile** [comment]
Yes, that could be an option, although I am not familiar with TypeScript yet 😅
I would be happy to do so if it helps other developers.

My understanding is that TypeScript is strongly typed so that each variable is assigned a specific type (I am sure there are other useful features). I tried my best to add jsDoc comments (and make a live documentation report) so that these are included in the current code base. 😊

**Reply by SeanPedersen** [reply]
Yes TypeScript adds explicit typing for vars. The migration can be done iteratively. Meaning you do not need to add all types add once but can do so over time. Not sure if I have the time to contribute the initial effort myself right now but I would recommend looking into it in the mid / long term for this projects.

**Reply by lbragile** [reply]
@SeanPedersen will do! Thank you for the input.

[Added](https://github.com/lbragile/TabMerger/projects/1#card-52528609) it to the todo list.

**Reply by lbragile** [reply]
@SeanPedersen I recently launched [v2.0.0](https://github.com/lbragile/TabMerger/releases/tag/v2.0.0) and so I have a bit of time before the next release is due. Started the migration to TypeScript - it is new to me - but so far I like it. Lots to learn. I will need to figure out how to make all the testing work still, how to build, and how to make the documentation. Thanks for suggesting this!
